kimchi/src/circuits/argument.rs

```diff
@@ -0,0 +1,23 @@
+//! An argument is simply a number of constraints,
+//! which we want to enforce on all points of the domain.
+//! Both the permutation and the plookup arguments fit this type.
+//! Gates can be seen as filtered arguments,
+//! which apply only in some points (rows) of the domain.
+
+use ark_ff::FftField;
+
+use crate::{
+    alphas::Alphas,
+    circuits::expr::{ConstantExpr, Expr},
+};
+
+/// The interface for a minimal argument implementation
+pub trait Argument {
+    type Field: FftField;
+
+    /// The number of constraints created by the argument
+    const CONSTRAINTS: usize;
+
+    /// Returns constraints safely combined via the passed combinator
+    fn constraint(&self, combinator: &Alphas<Self::Field>) -> Expr<ConstantExpr<Self::Field>>;
```

Comment:
Would it make more sense to return an `[Expr<ConstantExpr<Self::Field>>; Self::CONSTRAINTS]`? It seems like `constraint<T: Argument>(&self) -> Expr<..>` could be written once to combine them, rather than doing it multiple times times

Reply:
![image](https://user-images.githubusercontent.com/1316043/153685141-b59c80be-9bd7-42f1-9495-bb5dec42730a.png)

looks like const generics don't quite work yet, that's a bummer but we can fix that at some point in the future :o